### PR TITLE
rgw/notifications: automatically delete topics associated with a bucket

### PR DESCRIPTION
**Pull request by yuvalif**
fixes: https://tracker.ceph.com/issues/46128

> * ~this PR is depended with https://github.com/ceph/ceph/pull/38136. so, temporarily, this commit: https://github.com/ceph/ceph/commit/a4c906fc415d14849b89467634493e132c78a931 was added to this PR. please review without that commit. once https://github.com/ceph/ceph/pull/38136 is merged, the above commit will be removed from here.~
> * fixing the topic list (to exclude the auto-generated ones) will be done in a different PR

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/radosgw/notifications.rst [hunk 1]
```diff
@@ -50,25 +50,25 @@ In this case, the only latency added to the original operation is of committing
 Topic Management via CLI
 ------------------------
 
-Configuration of all topics of a user could be fetched using the following command:
+Configuration of all topics of (per tenant) could be fetched using the following command:
```
Comment (mattbenjamin) [inline doc/radosgw/notifications.rst:53]: accessible by a tenant?

**Review by mattbenjamin — APPROVED** [review]
lgtm

File: doc/radosgw/notifications.rst (hunk 1 above)
Comment (yuvalif) [reply doc/radosgw/notifications.rst:53]: this is actually a change from here: https://github.com/ceph/ceph/pull/38136
will fix there.

**Comment by yuvalif** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
jenkins test docs

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
jenkins test make check

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
jenkins test api

**Comment by yuvalif** [comment]
teuthology results: http://pulpito.front.sepia.ceph.com/yuvalif-2020-12-15_14:14:30-rgw-wip-yuval-fix-46128-distro-basic-gibba/
mostly green, with 4 failures:
- connection error. no crash dump found: http://qa-proxy.ceph.com/teuthology/yuvalif-2020-12-15_14:14:30-rgw-wip-yuval-fix-46128-distro-basic-gibba/5710341/teuthology.log
- duplicate pubsub events found when fetched via radosgw-admin (issue seen before)
- not sure what the cause here: http://qa-proxy.ceph.com/teuthology/yuvalif-2020-12-15_14:14:30-rgw-wip-yuval-fix-46128-distro-basic-gibba/5710331/teuthology.log
- `test_create_object_with_transfer_encoding` is failing: http://qa-proxy.ceph.com/teuthology/yuvalif-2020-12-15_14:14:30-rgw-wip-yuval-fix-46128-distro-basic-gibba/5710284/teuthology.log

**Comment by yuvalif** [comment]
another teuthology run: http://pulpito.front.sepia.ceph.com/yuvalif-2020-12-16_19:50:59-rgw-wip-yuval-fix-46128-distro-basic-gibba/
1 dead test and 2 failures:
- `PG_AVAILABILITY` issue
- slow OSDs: http://qa-proxy.ceph.com/teuthology/yuvalif-2020-12-16_19:50:59-rgw-wip-yuval-fix-46128-distro-basic-gibba/5713878/teuthology.log

**Pull request merged**